src/components/CustomIconButton.js

```diff
@@ -0,0 +1,56 @@
+import Icon from "./Icon";
+import React, { useState } from "react";
+import { Button, HStack, Box } from "@chakra-ui/react";
+
+const IconButton = (props) => {
+  const [isClicked, setIsClicked] = useState(false);
+
+  const handleClick = () => {
```

Comment:
This is bad practice since state is async. Use `setIsClicked(p => !p)` which uses the previous state